main.go

```diff
@@ -13,18 +16,104 @@ const (
 )
 
 type Item struct {
-	// title  string
-	// author string
-	// score  int
+	Title  string `json:"title"`
+	Author string `json:"by"`
+	Score  int    `json:"score"`
+	// time   time.Time
 	// absolutePostTime  time.Time
 	timeSincePosting time.Duration
-	URL              string
+	URL              string `json:"url"`
 }
 
 type Items []Item
 
+type HNItem struct {
+	id          json.Number
+	by          string
+	descendants json.Number
+	kids        []json.Number
+	score       json.Number
+	time        time.Time
+	url         string
+	title       string
+}
+
+type HNItems []HNItem
+
+var HItem Item
+var HItems Items
+var mf []interface{}
+
+var HNPost HNItem
+var HNFrontpage HNItems
+
+var f interface{}
+var fs []interface{}
+
+// var dat []interface{}
+
+// var test Item
+
 func main() {
 	fmt.Println("Hello hn-cli")
+
+	// item1 := Items{}
+	testJSON := Item{"Alice in Wonderland", "Bob", 666, 15966000000000000, "example.com/404"}
+
+	var storeData Item
+	marshallData, mErr := json.Marshal(testJSON)
+	if mErr != nil {
+		fmt.Println(mErr)
+	}
+	fmt.Println(string(marshallData))
+
+	unmarschallErr := json.Unmarshal(marshallData, &storeData)
+	if unmarschallErr != nil {
+		fmt.Println(unmarschallErr)
+	}
+	// marshallData == []byte(`{"Title":"Alice in Wonderland","Author":"Bob","TimeSincePosting":15966000000000000}`)
+	fmt.Println(storeData)
+
+	// var unmarshallData Item
+	// unmarshallErr := json.Unmarshal(marshallData, &unmarshallData)
+	// if unmarshallErr != nil {
+	// 	fmt.Println(unmarshallErr)
+	// }
+	// fmt.Println(unmarshallData)
+	// debug
+	// fmt.Println(test)
+	// testBytes := []byte(test)
+	// fmt.Println(string(testBytes))
+
+	// merr := json.Unmarshal(testBytes, &item1)
+	// if merr != nil {
+	// 	fmt.Println(merr)
+	// }
+	// fmt.Println(item1)
+
+	// marshallTest, mtErr := json.Marshal(Item{timeSincePosting: 15966000000000000})
+	// if mtErr != nil {
+	// 	panic(mtErr)
+	// }
+	// fmt.Println(string(marshallTest))
+
+	resp, err := http.Get("https://hacker-news.firebaseio.com/v0/topstories.json")
```

Comment:
This returns a JSON object that can be unmarshalled into `[]int`, not into `[]struct`.